features/signal/external/feature.go

```diff
@@ -0,0 +1,51 @@
+package external
+
+import (
+	"context"
+	"time"
+
+	"go.temporal.io/sdk-features/harness/go/harness"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/worker"
+	"go.temporal.io/sdk/workflow"
+)
+
+const (
+	SignalName = "external_signal_channel"
+	SignalData = "Signaled!"
+)
+
+var Feature = harness.Feature{
+	Workflows:       Workflow,
+	WorkerOptions:   worker.Options{MaxHeartbeatThrottleInterval: 1 * time.Nanosecond},
+	ExpectRunResult: SignalData,
+	Execute: func(ctx context.Context, runner *harness.Runner) (client.WorkflowRun, error) {
+		run, err := runner.ExecuteDefault(ctx)
+		if err != nil {
+			return nil, err
+		}
+		runner.Client.SignalWorkflow(
+			context.Background(),
+			run.GetID(),
+			run.GetRunID(),
+			SignalName,
+			SignalData,
+		)
+		return run, nil
+	},
+}
+
+// Workflow waits for a single signal and returns the data contained therein
+func Workflow(ctx workflow.Context) (string, error) {
+	wfResult := ""
+	signalCh := workflow.GetSignalChannel(ctx, SignalName)
+	selector := workflow.NewSelector(ctx)
```

Comment:
You don't need a selector with only one arm, you can just receive on the channel directly